generator.py

```diff
@@ -30,8 +30,13 @@ async def spawn_process(exe, args):
     return stdout, stderr
 
 
-async def spawn_gir(gir_exe, args):
-    stdout, stderr = await spawn_process(gir_exe, args)
+async def spawn_gir(conf, args):
+    if conf.gir_path:
+        stdout, stderr = await spawn_process(conf.gir_path, args)
+    else:
+        cargo_args = ["run", "--release", "--quiet", "--manifest-path=gir/Cargo.toml", "--"]
```

Comment:
Shouldn't that be `DEFAULT_GIR_DIRECTORY`? Now you're assuming that custom projects always have a submodule named `gir`? Perhaps we better fold this in with the configuration parameter `gir_path` somehow?

Reply:
> Shouldn't that be `DEFAULT_GIR_DIRECTORY`? 

Could make sense to change this. But I'm currently following the previous code where it was hardcoded.

> Now you're assuming that custom projects always have a submodule named `gir`? 

That's nothing new, it's the same as with the previous code:

```
def update_workspace():
    return run_command(["cargo", "build", "--release"], "gir")
```

> Perhaps we better fold this in with the configuration parameter `gir_path` somehow?

Well, that path is a binary path. Not sure how you would couple that with a source directory for gir.

Reply:
By adding a "defaulted" "gir directory" path argument, and making it mutually exclusive with `--gir-path` perhaps? That way the default (hardcoded) "gir" suddenly becomes visible too.